backend/fms_core/viewsets/sample.py

```diff
@@ -416,19 +417,17 @@ def versions(self, _request, pk=None):
 
     @action(detail=False, methods=["post"])
     def add_samples_to_study(self, request, pk=None):
-        excepted_sample_ids = request.data.get("excepted_sample_ids")
         default_selection = request.data.get("default_selection", False)
+        excepted_sample_ids = request.data.get("excepted_sample_ids", [])
         project_id = request.data.get("project_id")
         study_letter = request.data.get("study_letter")
         step_order = request.data.get("step_order", None)
 
         samples = self.filter_queryset(self.get_queryset())
+        samples = defaultSelection_exceptedIDs(samples, default_selection, excepted_sample_ids)
```

Comment:
The except option makes it that much harder to optimize the query ... Instead of limiting the query to X samples we have to work with N - X which likely forces a full scan of the table(and is slow as all hell). I am curious if only using positive selection (default_selection=false) would be faster. Might be worth investigating ...

Reply:
It's a trade-off between CPU and memory.